comps/dataprep/pinecone/prepare_doc_pinecone.py

```diff
@@ -1,40 +1,106 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+import json
 import os
+import shutil
+import uuid
+from pathlib import Path
+from typing import List, Optional, Union
 
 from config import EMBED_MODEL, PINECONE_API_KEY, PINECONE_INDEX_NAME
+from fastapi import Body, File, Form, HTTPException, UploadFile
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceEmbeddings, HuggingFaceHubEmbeddings
-from langchain_community.vectorstores import Pinecone
+from langchain_pinecone import PineconeVectorStore
+from langchain_text_splitters import HTMLHeaderTextSplitter
+from langsmith import traceable
+from pinecone import Pinecone, ServerlessSpec
 
 from comps import DocPath, opea_microservices, opea_telemetry, register_microservice
-from comps.dataprep.utils import document_loader, get_separators
+from comps.dataprep.utils import (
+    create_upload_folder,
+    document_loader,
+    encode_filename,
+    get_file_structure,
+    get_separators,
+    get_tables_result,
+    parse_html,
+    remove_folder_with_ignore,
+    save_content_to_local_disk,
+)
 
-tei_embedding_endpoint = os.getenv("TEI_ENDPOINT")
+tei_embedding_endpoint = os.getenv("TEI_EMBEDDING_ENDPOINT")
+upload_folder = "./uploaded_files/"
 
 
-@register_microservice(
-    name="opea_service@prepare_doc_pinecone",
-    endpoint="/v1/dataprep",
-    host="0.0.0.0",
-    port=6000,
-    input_datatype=DocPath,
-    output_datatype=None,
-)
-@opea_telemetry
-def ingest_documents(doc_path: DocPath):
+def check_index_existance(client):
+    print(f"[ check index existence ] checking {PINECONE_INDEX_NAME}")
+    pc = Pinecone(api_key=PINECONE_API_KEY)
+    existing_indexes = [index_info["name"] for index_info in pc.list_indexes()]
+    if PINECONE_INDEX_NAME not in existing_indexes:
+        print("[ check index existence ] index does not exist")
+        return None
+    else:
+        return True
+
+
+def create_index(client):
```

Comment:
The functions `check_index_existance`, `create_index` and `store_by_id` seems unused anywhere, could you please explain why put these in here?

Reply:
I have added them after looking into the Redis/qdrant files. But in case of pinecone, i have not used them. In future if we want to extend the functionality these can be used. Let me know.

Reply:
Unused functions are removed.